### ISSUE TITLE
User Registers his Business for first time. Business Registration.

### ISSUE THREAD

**Issue by amit-k86** [OrderMail/blacklabel#7]
### Description of Scenario’s:
At the time of Signing Up, User enters his Business Name. In case Business Name present the system sends the mail to the Administrator of that Business. 

Otherwise, In case of New Company User has to create a new company to proceed with the signing up.

Please note: Yesterday we discussed, a pop up will open once he enters a name of a new business. However I suggest rather than having pop up, when he clicks on the sign up button in the home page sign up form the new page to open with Business Registration or Mail Sent to Administrator whichever may be applicable to him.

Acceptance Criteria:
1.	User enter a name of a new business in the Sign Up form. When he clicks on the Sign Up Button after filling all details, He is taken to another Page known as business registration.
2.	Fields should be as follows:-
•	Business Name – Auto. Taken from the input field of the Sign up Form.
•	Address of Business – Can be Multiple and one has to be primary. Further details in the mock up.
•	Nature of Business – Keywords by user to be put.
•	Website
3.    Data should be available at the backend.
4.    Success Message to be displayed – “Business is successfully registered with Black Label. Kindly visit your Account Centre for further Setup Up Details” 
5. Or will be diverted to Account Centre. Details will be shared soon.

### SubTasks:
- [x] Design a Screen Wireframe
- [x]	Review the mock up.
- [x]	Create the Screen.
- [x]	Develop the Backend Service
- [x]	Integrate
- [x]	Writing Test Cases.

![mockup - bl business registration](https://cloud.githubusercontent.com/assets/9941976/5423254/ee8c59c6-82d8-11e4-8750-279dc4a3f823.png)

**Comment by priyankshukla** [comment]
I picked this story today as my environment got stable now.

**Comment by priyankshukla** [comment]
Created a new branch "business registration #7".
Committed code there, till this point of development.

**Comment by priyankshukla** [comment]
Created a new angular service, controller, model. Infect introduced a new custom module for Business registration.
I am able to register a business in new table called, Business in mongodb today.

**Comment by priyankshukla** [comment]
Code has committed.

It is accessible at http://localhost:3000/#!/businessregistration

**Comment by priyankshukla** [comment]
![business registration](https://cloud.githubusercontent.com/assets/6141437/5918780/931feee8-a652-11e4-9ba1-bb0b260aa996.PNG)

**Comment by vikasagarwal** [comment]
@priyankshukla: I checked out your branch and could access the business registration page. However, it's:
- Not saving the business after I enter details
- Appearing distorted unlike the screenshot you pasted up here

Are you sure that the story has been completed and the latest code has been committed? If not, please commit so that it can be reviewed and merged into master for Amit's review. I also need to have few businesses registered in order to start working on issue#12

**Comment by vikasagarwal** [comment]
@amit-k86, @priyankshukla, and @naveen1601: We also need to discuss if we should allow registration of a business only after a user logs in. In such a case, we won't be able to display the New Business Registration link at the time of User sign up. @amit-k86 Your help needed.

**Comment by priyankshukla** [comment]
@vikasagarwal, A successful message is appearing in current commit after registering a new business on Blacklabel.
Yes, the UI part is distorted as I made few changes after pasting the screenshot.

It's is implemented, and same you can verify by eod (I will be committing my code shortly).

**Comment by vikasagarwal** [comment]
Merged it locally on my machine so that I can start on issue #12

**Comment by vikasagarwal** [comment]
I just noticed, when I click the New Business Registration link on the User Sign-up page, it takes me a business registration page which is different from what Priyank is workingon as part of this story. Is it supposed to be same?

**Comment by priyankshukla** [comment]
@vikasagarwal, please ignore that link (on user registration page).

As per initial requirement, I pasted this link on user registration screen and that is still referring the business registration functionality. I have not disabled this as our discussion is still open on this!

Also, I created new module and designed a new form, business_registration_form, which has different height & width parameter from user_registration_form. I avoided changing user_registration_form size as user registration functionality would have get effected.

I will incorporate the changes, once we would freeze the requirement.

**Comment by priyankshukla** [comment]
Have just committed the code.. 
-Fixed UI issue

**Comment by priyankshukla** [comment]
Error handling implemented.
Code has committed on branch .

Pending task-
1. successful message should be in green colour
2. Needs to be decided, whether this functionality should be part of user registration or that needs to be accessible after successfully logged-in.

Note- We will create a separate story for above points, marking the story as dev-completed!

**Comment by vikasagarwal** [comment]
For now we can assume that this is available only after logging in and
ensuring that the person is activated

**Comment by vikasagarwal** [comment]
@priyankshukla : I pull your updates and merged local master into it. Here are my observations:
1. It's not displaying mandatory fields
2. The register button is grey which appears like it's inactive. It turns green when you hover mouse on it but that's not right. I think it should turn green when all the mandatory fields have been filled OR it should be green by default.
3. Are we agreeing that it should appear only for the logged in users? If so, I think it makes sense to complete this small change as part of this story itself? With that said, please feel free to create a new story if you think that's the right way.

**Comment by vikasagarwal** [comment]
@priyankshukla please pull your feature-businessregistration7 branch again before you start working on this branch again. I had merged master into this and made a minor refactoring as well.

**Comment by priyankshukla** [comment]
Thank you for highlighting @vikasagarwal.

I am going to pick/look and fixed (if required), the highlighted points 1 & 2 (above) in this story.
For #3, I will be implementing this in issue #16.

**Comment by priyankshukla** [comment]
![business registration new](https://cloud.githubusercontent.com/assets/6141437/6000554/32a9920a-ab05-11e4-943b-633a5b1c1e23.PNG)

**Comment by priyankshukla** [comment]
@vikasagarwal for code review!

**Comment by vikasagarwal** [comment]
QA done by me. Moving to QA Passed. Will be closed after merge with the master.

**Comment by priyankshukla** [comment]
@vikasagarwal 
I notice that the QA has been done but status of the ticket moved to 'In Review' from 'QA Passed', Is there any specific reason for that? I will check weather code has been merged to master or not,  this is required to closing this call..

**Comment by priyankshukla** [comment]
Had discussed with Vikas. 
He has highlighted few concerns in his code review.

Taken this back to 'in dev'

**Comment by priyankshukla** [comment]
@vikasagarwal I fixed all the issues which are highlighted in previous review.
Note- Biz registration html page has already been re-factored by you.

Please review the latest code, I pushed everything at feature-businessregistration-7 b ranch

**Comment by priyankshukla** [comment]
Review passed and tested.

Closing the story though!
